src/hdt.rs

```diff
@@ -57,6 +59,85 @@ impl Hdt {
         Ok(hdt)
     }
 
+    /// Creates an immutable HDT instance containing the dictionary and triples from the Path.
+    /// Will utilize a custom cached TriplesBitmap file if exists or create one if it does not exist.
+    /// The file path must point to the beginning of the data of an HDT file as produced by hdt-cpp.
+    /// FourSectionDictionary with DictionarySectionPlainFrontCoding and SPO order is the only supported implementation.
+    /// The format is specified at <https://www.rdfhdt.org/hdt-binary-format/>, however there are some deviations.
+    /// The initial HDT specification at <http://www.w3.org/Submission/2011/03/> is outdated and not supported.
+    /// # Example
+    /// ```
+    /// let hdt = hdt::Hdt::new_from_file(std::path::Path::new("tests/resources/snikmeta.hdt")).unwrap();
+    /// ```
+    #[cfg(feature = "cache")]
+    pub fn new_from_file(f: &std::path::Path) -> Result<Self, Box<dyn Error>> {
```

Comment:
As the function name is `new_from_file`, shouldn't it take a file instead of a path?